main.tf

```
@@ -421,3 +421,19 @@ resource "aws_s3_bucket_ownership_controls" "this" {
     aws_s3_bucket.this
   ]
 }
+
+resource "aws_s3_bucket_metric" "this" {
+  for_each = var.create_bucket ? { for m in var.bucket_metrics : m.name => m } : tomap({})
```

```suggestion
  for_each = { for k,v in var.bucket_metrics : k => v if var.create_bucket}
```

you'll need to update below for `name = each.value.name`

---

Ok it's modified.
I just modified to use the name as index, so we don't have to destroy/recreate each metric if you override a metric on top of the list of metrics.